### PR TITLE
docs(idempotency): fixes to testing your code section

### LINKED ISSUE #2072: Docs: Fix local tests example in the Idempotency Testing Your Code section
### What were you searching in the docs?

Improve the documentation by changing the example to be more realistic.

### Is this related to an existing documentation section?

Idempotency

### How can we improve?

Changing documentation

### Got a suggestion in mind?

_No response_

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #2072 

## Summary

Fix local test example

### Changes

 - Changed documentation

### User experience

Before
image

After
image

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Review by rubenfonseca — APPROVED** [review]

File: docs/utilities/idempotency.md
Comment (heitorlessa) [inline docs/utilities/idempotency.md]: have you tested it? sounds like we're missing something here - we're creating a low-level client, not resource client anymore, but then we don't use it... and we're using a hardcoded table name instead of using the previously registered table name in the persistency layer.

Comment (leandrodamascena) [reply docs/utilities/idempotency.md]: I tested it and was able to reproduce it in my local environment, even when it fails because I added a table that doesn't exists. But yes, it might be good if you double-check.

I added the hardcoded name to demonstrate to the customer that it can be changed by setting a new value for the variable. Doesn't it make sense for us to keep this or even get it from the current `app.persistence_layer.table_name` variable? I think just removing the hardcoded we don't show the real use of this.

Comment (heitorlessa) [reply docs/utilities/idempotency.md]: Definitely get from the current `table_name` variable (test to be sure). Reason being is that we're trying to demonstrate how customers would test themselves - if they use Serverless framework for example, a plugin would automatically create a DynamoDB table with the name of the table they're using in Lambda.

If we show it hard coded, it could lead to confusion - `Wait? Do I have to use a different table name? Should I update my DynamoDB Local table bootstrap script?`.

Docs point of view: We're testing `app.py` who already set the table name, `DynamoDBPersistenceLayer(table_name="idempotency")`. From a testing point of view, you'd want to use the same resource defined there, so you don't get a drift. If you want to set a separate one, a good practice would be to actually set the table name in `app.py` using env var which a test could inject before importing it :) 

Hope that makes sense.

Comment (leandrodamascena) [reply docs/utilities/idempotency.md]: > Definitely get from the current table_name variable (test to be sure). Reason being is that we're trying to demonstrate how customers would test themselves - if they use Serverless framework for example, a plugin would automatically create a DynamoDB table with the name of the table they're using in Lambda.

Done. I added a comment to explain that the user can change the variable value and use other value for the table name.

> If we show it hard coded, it could lead to confusion - Wait? Do I have to use a different table name? Should I update my DynamoDB Local table bootstrap script?.

Yes, I totally agree with your explanation.

> Docs point of view: We're testing app.py who already set the table name, DynamoDBPersistenceLayer(table_name="idempotency"). From a testing point of view, you'd want to use the same resource defined there, so you don't get a drift. If you want to set a separate one, a good practice would be to actually set the table name in app.py using env var which a test could inject before importing it :)

Including this as an environment variable might confuse users that this environment variable is something from the idempotency provider and not specific to this test, no? I mean, for example, we have a `POWERTOOLS_IDEMPOTENCY_DISABLED` variable that is global to the provider and it makes sense to set it via an environment variable. I'm really not sure, I'm just bringing one more thought to the table.

**Comment by leandrodamascena** [comment]
I am moving the comment from issue to PR. 
Refer: https://github.com/awslabs/aws-lambda-powertools-python/issues/2072#issuecomment-1494339573

> Heitor Lessa:
only last comment before merging is that this part shouldn't work, please double check in case I'm wrong: [#2073 (files)](https://github.com/awslabs/aws-lambda-powertools-python/pull/2073/files#diff-e612f95a3dbd33cabab2e49899653b8cc5d5e809bfad45aa54cdee41d22e5808R1253)
Reason. Unless AWS SDK changed, the DynamoDB client wouldn't connect to DynamoDB Local because the client isn't being replaced, therefore it'd use the real DynamoDB external endpoint

Yeah @heitorlessa, you're right! I was missing it because in my local tests, I was configuring the boto3.session in the `app.py` file and I forgot to remove it. My bad. I already sent the new code.

**Comment by heitorlessa** [comment]
@leandrodamascena created PR exposing the SDK Client **and** accepting a custom SDK client so customers can do this more cleanly.

I'll let you review, once merged, then we can finish this one.

https://github.com/awslabs/aws-lambda-powertools-python/pull/2087

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**
